File: Heap/last-stone-weight.java
import java.util.*;
import java.util.Collection;
import java.util.Collections;
import java.util.PriorityQueue;

class Solution  {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder()); 
        for(int ele : stones){
            pq.add(ele);
        }
        while(pq.size()>1){
            int last = pq.remove();
            int slast = pq.remove();
            if(last!=slast) pq.add(last-slast);
        }
        return (pq.size()!=0 ? pq.remove() : 0);      
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        int[] arr = {2,7,4,1,8,1};
        System.out.println(s.lastStoneWeight(arr));
    }
}

File: Heap/last-stone-weight.js
var lastStoneWeight = function(stones) {
    while (stones.length>1) {
        stones.sort((a,b)=> a-b);
        let last = stones.pop();
        let slast = stones.pop();
        if(last!=slast) stones.push(last-slast);
    }
    return (stones.length>0 ? stones[0] : 0);
};

// let stones = [2,7,4,1,8,1];
let stones = [2,2];

console.log(lastStoneWeight(stones));